src/shared/install.c

```diff
@@ -2286,10 +2297,14 @@ static int install_context_mark_for_removal(
                         }
                 } else if (r < 0) {
                         log_debug_errno(r, "Failed to find unit %s, removing name: %m", i->name);
-                        install_changes_add(changes, n_changes, r, i->path ?: i->name, NULL);
+                        r = install_changes_add(changes, n_changes, r, i->path ?: i->name, NULL);
+                        if (r < 0)
+                                return r;
```

Comment:
Maybe, something like the following?
```c
int k = install_changes_add(...);
if (k != r)
        return k;
```
Previously, the symlink is removed in the below, and the loop is continued. But with this change, the function returns here and remaining entries will be ignored.